internal/authproxy/serve.go

```diff
@@ -132,6 +144,37 @@ func Command() (*cobra.Command, error) {
 	return serveCmd, nil
 }
 
+type (
+	RequestStats struct {
+		LastRequest time.Time
+		mutex       sync.RWMutex
+	}
+)
```

Comment:
Nitpick: What you have is fine, it is just unnecessary.

```suggestion
type RequestStats struct {
	LastRequest time.Time
	mutex       sync.RWMutex
}

```